### PR TITLE
Azure Pipelines: Replace macOS 10.13 with 10.15

### PR DESCRIPTION
**Pull request by raimue**
## Description

macOS-10.13 has been deprecated for Azure Pipelines and will be removed
soon. Replace it with the newly added macOS-10.15 instead.

Are we ready to run the tests on macOS 10.15 or do we still need to provide something for the bootstrap script?

###### Type(s)

- [ ] bugfix
- [x] enhancement
- [ ] security fix

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL?

- [ ] checked your Portfile with `port lint`?
- [ ] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?

### CODE REVIEW

**Comment by raimue** [comment]
Well, I guess the build failure on this pull request shows that we are not ready yet... :-)

```
curl: (22) The requested URL returned error: 404 Not Found
```

I conclude this is because `MacPorts-${OS_MAJOR}.tar.bz2` is missing on bintray. Maybe we should use `set +x` in the bootstrap script to see that immediately...

@l2dy Any hints on how this is tarball has to be built and how it can be published?

**Comment by pmetzger** [comment]
It doesn't work of course, but in principle I'm 100% in favor of progressing this. Azure Pipelines are de facto our CI system right now. Travis really doesn't work most of the time.

**Comment by l2dy** [comment]
> Well, I guess the build failure on this pull request shows that we are not ready yet... :-)
> 
> ```
> curl: (22) The requested URL returned error: 404 Not Found
> ```
> 
> I conclude this is because `MacPorts-${OS_MAJOR}.tar.bz2` is missing on bintray. Maybe we should use `set +x` in the bootstrap script to see that immediately...
> 
> @l2dy Any hints on how this is tarball has to be built and how it can be published?

They are built by Travis CI in the [travis-ci](https://github.com/macports/macports-base/tree/travis-ci) branch in macports-base. Unfortunately, Travis CI does not support macOS 10.15 yet. https://docs.travis-ci.com/user/reference/osx/#macos-version

**Comment by reneeotten** [comment]
Travis CI does [support macOS 10.15 now](https://docs.travis-ci.com/user/reference/osx/#macos-version), so this should probably be revisited?

**Comment by reneeotten** [comment]
I see that macOS 10.15 (`osx_image: xcode11.4`) was added to the `travis_ci` branch mentioned above. However, that build fails so the `MacPorts-${OS_MAJOR}.tar.bz2` is still missing. The failure seems as described in [Ticket 59697](https://trac.macports.org/ticket/59697), with commits from @jmroot to fix it (if I read it correctly).

Perhaps this can be fixed then by just adding these commits to the `travis_ci` branch? It would be nice to be able to test PRs against macOS 10.15.

**Comment by jmroot** [comment]
There's still at least one more test failing on 10.15 BTW.

**Comment by reneeotten** [comment]
can we just disable the failing tests for now to get this working? There are installers for Catalina and it appears that MacPorts works fine there; apparently the failing tests were not a problem for making an installer, so it cannot be that broken - or am I missing something here?

**Comment by neverpanic** [comment]
Wasn't there a recently a PR on base to fix the failing test on 10.15? How current is your check that it fails?

**Comment by reneeotten** [comment]
now that 2.6.3 is released (thanks for that!) can someone fix the `travis_ci` branch in `macports-base`? Or, alternatively, add the required changes in `.travis.yml` and ` .bintray.json` once a new version is tagged, so in this case to the `2.6.3` branch? I am not sure what is deemed the best way forward; it would be nice however to get this working.

**Comment by chrstphrchvz** [comment]
> now that 2.6.3 is released (thanks for that!) can someone fix the `travis_ci` branch in `macports-base`?

This is done now: macports/macports-base@71afd037b7

**Review by neverpanic — APPROVED** [review]

**Comment by neverpanic** [comment]
I guess we don't need to wait for the Azure CI on this PR. I'll see if I can test this immediately after merging with #8153.

**Comment by neverpanic** [comment]
And confirmed working.

**Pull request merged**
